source/Hardpoint.cpp

```diff
@@ -178,58 +178,66 @@ void Hardpoint::Step()
 		return;
 
 	wasFiring = isFiring;
-	if(reload > 0.)
-		--reload;
-	// If the full reload time is elapsed, reset the burst counter.
-	if(reload <= 0.)
-		burstCount = outfit->BurstCount();
-	// If the weapon recently fired then increment any sustained fire counters.
 	if(reload > 0.)
 	{
+		--reload;
+		// If the weapon recently fired then increment any sustained fire counters.
 		if(bloomCount < outfit->BloomTime())
 			++bloomCount;
 		if(spinupCount < outfit->SpinupTime())
 			++spinupCount;
 	}
-	// Only decrement sustained fire counters if this weapon could have fired
-	// but didn't.
-	else if(!wasFiring)
+	if(reload <= 0.)
```

Comment:
Why isn't this an else?
```suggestion
	else
```